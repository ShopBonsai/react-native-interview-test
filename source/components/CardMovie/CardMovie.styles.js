import styled from 'styled-components'
import { View, Image } from 'react-native'

export const CardMovieContainer = styled(View)`
  flex: 1;
`
export const CardMovieContent = styled(View)`

`
export const CardMovieImage = styled(Image)`
  width: 120px;
  height: 150px;
  background-color: #cccccc;
  border-radius: 5px;
  margin: 10px;
`
