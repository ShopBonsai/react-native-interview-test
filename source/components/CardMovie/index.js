import React from 'react'
import PropTypes from 'prop-types'
import { CardMovieContainer, CardMovieContent, CardMovieImage } from './CardMovie.styles'

/**
 * The CardMovie it's responsible by to render the
 * card with an image (thumnal)
 *
 * @export {FUNCTION}
 * @param {Object} props
 * @returns {JSX.Element}
 */
export default function CardMovie(props) {
  // properties
  const { thumbnail } = props

  // main render
  return (
    <CardMovieContainer>
      <CardMovieContent>
        <CardMovieImage
          source={{ uri: thumbnail }}
        />
      </CardMovieContent>
    </CardMovieContainer>
  )
}

// define the default properties
CardMovie.defaultProps = {
  thumbnail: 'https://via.placeholder.com/120x150'
}

// validate the default properties
CardMovie.propTypes = {
  thumbnail: PropTypes.string
}
