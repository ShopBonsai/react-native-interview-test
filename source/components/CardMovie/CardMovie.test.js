// libraries
import React from 'react'
import renderer from 'react-test-renderer'
import CardMovie from './index'

// components

it('Renders CardMovie correctly', () => {
  const tree = renderer
    .create(<CardMovie />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
