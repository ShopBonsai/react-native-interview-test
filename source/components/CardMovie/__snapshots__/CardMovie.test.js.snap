// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CardMovie correctly 1`] = `
<View
  className="sc-bdVaJa gLHRUG"
>
  <View
    className="sc-bwzfXH fNxhoP"
  >
    <Image
      className="sc-htpNat joWonT"
      source={
        Object {
          "uri": "https://via.placeholder.com/120x150",
        }
      }
    />
  </View>
</View>
`;
